recipe/meta.yaml

```
@@ -59,21 +64,30 @@ requirements:
     # Pin all LLVM and clang packages to the latest version with each build.
     # This ensures that HOOMD can find the corresponding header files which
     # conda installs to version specific directories.
-    - llvmdev 14.0.6
-    - clangdev 14.0.6
+    - llvmdev 15.0.7  # [not linux64]
+    - clangdev 15.0.7  # [not linux64]
+    - llvmdev 14.0.6  # [linux64]
+    - clangdev 14.0.6  # [linux64]    
 
   run:
     - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx and x86_64]
     - __cuda >={{ cuda_compiler_version }}  # [linux64 and cuda_compiler_version != "None"]
+    - cuda-nvrtc  # [linux64 and cuda_compiler_version != "None"]
+    - libcufft  # [linux64 and cuda_compiler_version != "None"]
+    - libcusolver  # [linux64 and cuda_compiler_version != "None"]
+    - libcusparse  # [linux64 and cuda_compiler_version != "None"]
```

These libraries are only available for CUDA 12. CUDA 11 will still require `cudatoolkit`. We might need:
```suggestion
    - cudatoolkit  # [linux64 and cuda_compiler_version.startswith("11")]
    - cuda-nvrtc   # [linux64 and cuda_compiler_version.startswith("12")]
    - libcufft     # [linux64 and cuda_compiler_version.startswith("12")]
    - libcusolver  # [linux64 and cuda_compiler_version.startswith("12")]
    - libcusparse  # [linux64 and cuda_compiler_version.startswith("12")]
```

---

This doesn't work:
```
Error: Invalid selector in meta.yaml line 74:
offending line:
    - cudatoolkit  # [linux64 and cuda_compiler_version.startswith("11")]
exception:
'bool' object has no attribute 'startswith'
```